docs/DeveloperGuide.md

```diff
@@ -13,47 +20,73 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
-
 ### UI Component
+
 Represented by the Ui class, it handles the interactions between the user and the program. In addition, it also displays
-a basic overview of what the program does. 
+a basic overview of what the program does.
 
-Types of pre-defined messages: 
+Types of pre-defined messages:
 
 1. Information messages that describe the functionalities of the program (`Greet` , `Bye`, `PrintExercises` classes)
 2. Help messages that details the usage of the possible commands available in the program (`PrintHelpMessage` class)
-3. Error messages that trigger when a user's input is incorrect and provides an explanation to the user. (`ErrorMessages` class)
+3. Error messages that trigger when a user's input is incorrect and provides an explanation to the
+   user. (`ErrorMessages` class)
 
-Written by : Glenn Ho
+### Storage Component
 
-### Exceptions
-Accounts for the different scenarios that may trigger an error during user's interactions with the program
+API: [```StorageHandler.java```](https://github.com/AY2223S2-CS2113-W13-2/tp/blob/master/src/main/java/seedu/duke/storage/StorageHandler.java)
 
-Written by : Glenn Ho
+The *Storage* component handles the reading and writing of user data to and from the local hard disk in the form of
+a json file.
 
+Key Aspects:
 
+* Handles the serialization and deserialization of user data into a json file using
+  the [Gson Library](https://github.com/google/gson)
+* Handles the creation of user data file when previous one is missing or corrupted
 
-### Additional features to be added
-1. An IPPT calculator to integrate with Fitness Duke to facilitate users who want to train for their IPPT fitness assessments.
+The class diagram as shown in *Figure X.1* illustrates the structre of the different classes in Storage.
+<div align="center">
+<img src="UML/Images/StorageHandlerClass.png" />

```

Comment:
Very detailed UML diagram to show the relationships between the methods and functions